x-pack/winlogbeat/module/security/ingest/security.yml

```diff
@@ -3442,6 +3458,51 @@ processors:
           splitSidList(ctx.winlog.event_data.SidList, params, ctx);
         }
 
+  - set:
+      field: file.name
+      copy_from: winlog.event_data.RelativeTargetName
+      if: |-
+        ctx.event?.code != null &&
+        ["5140", "5145"].contains(ctx.event.code) &&
+        ctx.winlog?.event_data?.RelativeTargetName != null &&
+        ctx.winlog?.event_data?.RelativeTargetName != ""
+  - set:
+      field: file.directory
+      copy_from: winlog.event_data.ShareLocalPath
+      if: |-
+        ctx.event?.code != null &&
+        ["5140", "5145"].contains(ctx.event.code) &&
+        ctx.winlog?.event_data?.ShareLocalPath != null &&
+        ctx.winlog?.event_data?.ShareLocalPath != ""
```

Comment:
```suggestion
        ctx.winlog.event_data.ShareLocalPath != ""
```

Reply:
These were all just a direct copy/paste from the Security Integration, but I agree the field checks are not needed since already verified.